src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java

```diff
@@ -217,13 +229,88 @@ private AuthConfig createStandardAuthConfig(boolean isPush, Map authConfigMap, S
             log.debug("AuthConfig: credentials from ~/.m2/setting.xml");
             return ret;
         }
+        
+        // check EC2 instance role if registry is ECR
+        if (EcrExtendedAuth.isAwsRegistry(registry)) {
+        	try {
+				ret = getAuthConfigFromEC2InstanceRole();
+        	} catch (ConnectTimeoutException ex) {
+				log.debug("Connection timeout while retrieving instance meta-data, likely not an EC2 instance (%s)",
+						ex.getMessage());
+        	} catch (IOException ex) {
+				// don't make that an error since it may fail if not run on an EC2 instance
+				log.warn("Error while retrieving EC2 instance credentials: %s", ex.getMessage());
+			}
+            if (ret != null) {
+                log.debug("AuthConfig: credentials from EC2 instance role");
+                return ret;
+            }        	
+        }
 
         // No authentication found
         return null;
     }
 
     // ===================================================================================================
-
+    
+    
+    // if the local credentials don't contain user and password, use EC2 instance role credentials
+    private AuthConfig getAuthConfigFromEC2InstanceRole() throws IOException {
+		log.debug("No user and password set for ECR, checking EC2 instance role");
+        try (CloseableHttpClient client = HttpClients.custom().useSystemProperties().build()) {
+        	// we can set very low timeouts because the request returns almost instantly on an EC2 instance
+        	// on a non-EC2 instance we can fail early
+        	RequestConfig conf = RequestConfig.custom().setConnectionRequestTimeout(1000)
+        			.setConnectTimeout(1000)
+        			.setSocketTimeout(1000).build();
+        	
+        	
+        	// get instance role - if available
+        	HttpGet request = new HttpGet("http://169.254.169.254/latest/meta-data/iam/security-credentials");
+        	request.setConfig(conf);
+        	String instanceRole;
+        	try (CloseableHttpResponse response = client.execute(request)) {
+        		if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+        			// no instance role found
+					log.debug("No instance role found, return code was %d",
+							response.getStatusLine().getStatusCode());
+        			return null;
+        		}
+        		
+        		// read instance role
+                try (InputStream is = response.getEntity().getContent()) {
+                	instanceRole = IOUtils.toString(is, StandardCharsets.UTF_8);
```

Comment:
any validation of this instance role needed ?

Reply:
What exactly do you mean with "validation" in this context?